### PR TITLE
Adds TypeScript declarations

### PR DESCRIPTION
**Pull request by vatson**
Recently, TS is becoming more popular. It would be great if you allow to add support for TypeScript.

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare module "vue-prism-component" {
+  import { FunctionalComponentOptions } from 'vue';
+  const _default: FunctionalComponentOptions;
+  export default _default;
+}
```
Comment (egoist) [inline types/index.d.ts]: Can you add types for props as well?

**Comment by vatson** [comment]
@egoist I added prop types, pls check

**Comment by phildebr** [comment]
Could you please merge this?

**Comment by mnapoli** [comment]
Would be great to have!

Edit: I can confirm that this works.

**Comment by rbclark** [comment]
Unsure if this is still up-to-date but came to this repo explicitly looking for typings for this package, would really be great to see this merged.

**Comment by Shinigami92** [comment]
> Unsure if this is still up-to-date but came to this repo explicitly looking for typings for this package, would really be great to see this merged.

Same

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+declare module "vue-prism-component" {
+  import { FunctionalComponentOptions } from 'vue';
+  const _default: FunctionalComponentOptions<{
```
Comment (Shinigami92) [inline types/index.d.ts]:
```suggestion
  import { FunctionalComponent } from 'vue';
  const _default: FunctionalComponent<{
```

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    code: string;
+    language: string;
```
Comment (Shinigami92) [inline types/index.d.ts]:
```suggestion
    code?: string;
    language?: string;
```

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Comment by elevatebart** [comment]
If this PR is only missing the changes suggested, can @vatson make them so we can merge? 
@egoist, @Shinigami92 what are your suggestions?

I could also fork @vatson repo, do the changes, and create a new PR.

**Comment by elevatebart** [comment]
This closes #39

**Comment by Shinigami92** [comment]
@vatson What's the status of this? Why just closed?

**Comment by vatson** [comment]
> @vatson What's the status of this? Why just closed?

@Shinigami92 

First, the PR has been stuck for 3 years. I don't see any interest and activity from the maintainer side. All of this tells me that this issue will be here forever.

Second, current `master` branch  is far ahead. It's has already been migrated on vue 3. It makes incompatible code with proposed typings.

I could do typings for both vue 2 and vue 3 with new PRs, but still I'm not sure if the project maintainers are ready to assist and support. It looks like a waste of time to me.

**Comment by Shinigami92** [comment]
@vatson In such a case, did you consider https://github.com/DefinitelyTyped/DefinitelyTyped ?

**Pull request closed without merging**